### ISSUE TITLE
Bug: indent rule not working with optional chaining

### ISSUE THREAD

**Issue by william-fargues** [eslint/eslint#15979]
### Environment

Node version: 14.15.5
npm version: 6.14.11
Local ESLint version: 7.6.0
Operating System: Windows

### What parser are you using?

@typescript-eslint/parser

### What did you do?

Normal situation: 
```js
let foo = 50.00
foo.toFixed(2)
```
In this scenario everything is working fine. 
So, I can't do this because I'll have the indent warning on the 2nd line:
```js
let foo = 50.00
            foo.toFixed(2)
```
But as soon as I add an optional chaining operator the indent rule is not working anymore.
Eslint take the following as valid (I don't have the warning anymore):
```js
let foo = 50.00
            foo?.toFixed(2)
```

### What did you expect to happen?

To have an eslint warning because the indentation is not correct on the 2nd line.

### What actually happened?

Nothing

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by aladdin-add** [comment]
what's your eslint config? To work with ts, you'll need to the alternative rule: `@typescript-eslint/indent`.

**Comment by william-fargues** [comment]
Here's my config:
```js
rules: {
    'semi': [1, 'never'],
    'quotes': [1, 'single'],
    'no-unused-vars': [0, { 'args': 'all' }],
    'object-curly-spacing': [1, 'always'],
    'space-before-function-paren': [1, 'never'],
    'space-infix-ops': [1, { 'int32Hint': false }],
    'comma-spacing': [1, { 'before': false, 'after': true }],
    'keyword-spacing': [1, { 'before': true }],
    'comma-dangle': [1, 'always-multiline'],
    'padded-blocks': [1, { 'classes': 'always', 'blocks': 'never' }],
    'spaced-comment': [1, 'always'],
    'indent':  [1, 2, { 'SwitchCase': 1 }],
    // '@typescript-eslint/indent': [1, 2, { 'SwitchCase': 1 }],
    'space-in-parens': [1, 'never'],
    'space-before-blocks': [1, 'always'],
    'key-spacing': [1, { 'beforeColon': false, 'afterColon': true }],
    'arrow-spacing': [1, { 'before': true, 'after': true }],
    'import/order': 1,
  },
  ```
  
  Yes, I have tried the ```@typescript-eslint/indent``` rule and it's the same

**Comment by aladdin-add** [comment]
still cannot repro it tho: https://typescript-eslint.io/play/#ts=4.7.2&sourceType=module&code=DYUwLgBAZg9jEF4IFYAMA6VqBQE-4L1hgH50wYAxASwA8QATACgCYBKIA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUFIAEAuAngA4oDG0AlkXgLQrwUB2eA9EwCaLPpgDaAjABoATIOBQAygHcKeUgAsAwgENkiHvwC+AXXBhNITUA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

can you upgrade eslint & ts-eslint to latest, and retry?

**Comment by william-fargues** [comment]
I did it and it worked thank you !
